AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1901,9 +1912,13 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
     layoutSpec.isMutable = NO;
     ASLayout *layout = [layoutSpec measureWithSizeRange:constrainedSize];
     // Make sure layoutableObject of the root layout is `self`, so that the flattened layout will be structurally correct.
-    if (layout.layoutableObject != self) {
+    BOOL isRootLayout = (layout.layoutableObject != self);
+    if (isRootLayout) {
```

Comment:
Was looking at this the other day. In what case will we get an ASLayout returned from the layoutSpec where the layoutableObject is self? Seems like only via returning `self` from ASLayoutSpec that fits.